gratipay/utils/__init__.py

```diff
@@ -132,6 +133,46 @@ def get_team(state):
     return team
 
 
+def b64encode_s(s):
+    prefix = b''
+    if not isinstance(s, bytes):
+        s = s.encode('utf8')
+    else:
+        # Check whether the string is binary or already utf8
+        try:
+            s.decode('utf8')
+        except UnicodeError:
+            prefix = b'.'
+    return prefix + b64encode(s, b'-_').replace(b'=', b'~')
+
+
+def b64decode_s(s, **kw):
+    def error():
+        if 'default' in kw:
+            return kw['default']
+        raise Response(400, "invalid base64 input")
+
+    try:
+        s = s.encode('ascii')
+    except UnicodeError:
+        return error()
+
+    udecode = lambda a: a.decode('utf8')
+    if s[:1] == b'.':
+        udecode = lambda a: a
+        s = s[1:]
+    s = s.replace(b'~', b'=')
+    try:
+        return udecode(b64decode(s, '-_'))
+    except Exception:
+        try:
+            # For retrocompatibility
+            return udecode(b64decode(s))
+        except Exception:
+            pass
+        return error()
+
+
```

Comment:
Examining some use cases([1](https://github.com/liberapay/liberapay.com/blob/f624590da031943dda62c3ee99b4e2a045bca55a/templates/base.html#L111), ([2](https://github.com/liberapay/liberapay.com/blob/6e98bd14f41a47b8cd9c146a4136e65f4dc02aa2/www/%25username/wallet/payin/%25back_to.spt#L78), [2,1](https://github.com/liberapay/liberapay.com/blob/6e98bd14f41a47b8cd9c146a4136e65f4dc02aa2/www/%25username/wallet/payin/%25back_to.spt#L120)), [3](https://github.com/liberapay/liberapay.com/blob/19ac142d7ee2360b1371c2fae2c5c3f4b8c84552/www/%25username/wallet/payout/%25back_to.spt#L25)), looks like `default` was used as a cleaner way to handle errors and avoid the long `try/except` block.